### PR TITLE
[DOC] Correctly mark public link-to component named arguments

### PR DESCRIPTION
**Pull request by bertdeblock**
Based on [this RFC summary](https://github.com/emberjs/rfcs/blob/modernizing-built-in-components/text/0707-modernize-built-in-components-2.md#summary) and [this part of the API docs](https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/LinkTo?anchor=LinkTo).

cc @chancancode.

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/components/link-to.ts [hunk 1]
```diff
@@ -406,6 +406,18 @@ const LinkComponent = EmberComponent.extend({
   **/
   replace: false,
 
+  /**
+    Determines whether the `LinkComponent` will prevent the default
```
Comment (bertdeblock) [inline packages/@ember/-internals/glimmer/lib/components/link-to.ts:410]: Based on the description of the `Allowing Default Action` section in [the LinkTo API docs](https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/LinkTo?anchor=LinkTo).

Comment (chancancode) [reply packages/@ember/-internals/glimmer/lib/components/link-to.ts:410]: Now that I think about it, I wonder if this should just be deprecated and removed as part of RFC 707.

This seems only useful in the context of changing the tagName to something other than `<a>` (which is deprecated in RFC 707).

When the tagName is `<a>`, the default action of the browser is to navigate away, so there is no sense in not preventing it.

Thoughts?

Looping in @rwjblue as well.

Comment (bertdeblock) [reply packages/@ember/-internals/glimmer/lib/components/link-to.ts:410]: I'm not sure I completely understand what you mean. I've never used `@preventDefault` myself, but I can imagine it being useful for some use cases to open a route in a new tab.

Comment (bertdeblock) [reply packages/@ember/-internals/glimmer/lib/components/link-to.ts:410]: Wow sorry, scratch that. I was confused. Yeah I can't immediately imagine a scenario where a refresh is desired.

Comment (rwjblue) [reply packages/@ember/-internals/glimmer/lib/components/link-to.ts:410]: @chancancode - ya seems correct to me

**Review by rwjblue — COMMENTED** [review]
@chancancode - So... Merge now and update the RFC RE: deprecating `preventDefault`?

**Comment by chancancode** [comment]
@rwjblue @bertdeblock How about we move the docs currently in the "Allowing Default Action" section into the documentation for the property itself, and mark the property as `@private`? That way the documentation can still technically be found if someone needs it but we avoid marking it as `@public` just to have it deprecated soon after.

Reading the docs again – I suppose there is case to be made for keeping this even if `@tagName` is deprecated. In theory, you may want a link to act like an "ember link" most of the time, except when certain condition is true, _then_ you want it to actually trigger a full page reload. Something like this:

```hbs
<LinkTo @route="dashboard" @preventDefault={{this.requiresRefresh}}>
  Return to dashboard
</LinkTo>
```

Where `this.requiresRefresh` is conditionally true, perhaps when the user is a new sign up that just finished the on boarding process, something like that.

If someone wants to make the case for it, I think it wouldn't cost much to keep the functionality around. Personally, I think it's not really needed, and there are probably better ways of accomplishing that without using `<LinkTo>`, so I am still leaning towards deprecating it, but I could go either way on that.

**Comment by chancancode** [comment]
I opened a PR on the RFC repo to amend this if we want to go forward: https://github.com/emberjs/rfcs/pull/710

**Comment by bertdeblock** [comment]
This sounds good to me. I don't have any incentive to keep this functionality around, but of course maybe other people do. Let me know when / if I should update this PR to move the docs 👍.

**Comment by chancancode** [comment]
I think you can go ahead, if we ended up wanting to make it public it's easy enough to change the tag.

**Comment by bertdeblock** [comment]
Moved the docs (hopefully I understood it correctly). Let me know if anything.

**Review by chancancode — APPROVED** [review]
@rwjblue seems good to you?

**Comment by bertdeblock** [comment]
Gentle ping. Anything left I should take care of?

**Review by chancancode — APPROVED** [review]
Sorry for the delay!

**Pull request merged**